### ISSUE TITLE
Implement Multi-Task deep learning with Tensorflow

### ISSUE THREAD

**Issue by MammadTavakoli** [nlintz/TensorFlow-Tutorials#42]
How can implement Multi-Task deep learning with Tensorflow? how can i found any sample?  ![Multi-Task deep learning](https://camo.githubusercontent.com/9b5b091d632c5b923b706016ca15ee56e70c9958/687474703a2f2f322e62702e626c6f6773706f742e636f6d2f2d74736736742d63547857302f56626e31416c51776e72492f414141414141414143546b2f487a5548794e62547478492f73313630302f53637265656e25324253686f74253242323031352d30372d33302532426174253242253235453125323538342532353842253235453125323538352532354139253235453125323538342532353932253235453125323538352532354145253242362e35352e30312e706e67)

**Comment by nlintz** [comment]
I think that you can simply grab out the features from the final fully conv and create separate weight matrices for each of the outputs. You'll need to make sure to set scope.reuse_variables() to equal true or else tensorflow won't let you reuse features.

**Comment by DoktorMoreau** [comment]
Hello, 

I also try to implement the 'Multi Task Learning' (MTL) paradigm with Tensorflow. I would like to use a 'Multilayer Perceptron' or an 'Convolutional Neural Network'. I use a faces dataset, with labels 'gender' and 'age'. These would be my two tasks.
- The faces image is about 450x450 pixels.
- The Gender is a one hot-vector of dimension 2. (male or female)
- The Age is a one hot-vector of dimension 2. (age over 40, or not)

The MTL approach consist of two kinds of parameter:
- Generic parameters: across all the tasks, lower layers of the network.
- Task-specific parameters: upper layers of the network.

So, how do I get two kinds of labeled data into the net? should I concatenate the two vectors? And how do I "split" the neuron and create a label-specific neuron and get two outputs from the net? How can I reuse variables?
Please can you give me some hint, or an code example?

I am very grateful if you could give me some hints.
Thank you very much.
